engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetInputTexture2D.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.stateChanges;
+
+import org.terasology.assets.ResourceUrn;
+
+import static org.lwjgl.opengl.GL11.GL_TEXTURE_2D;
+
+public class SetInputTexture2D extends SetInputTexture {
```

Comment:
The assumption with the SetInputTexture2D/3D classes is that developers will use these rather than the superclass, to keep node constructors less cluttered. As such, the javadoc of these two classes should be a near copy of the superclass, adapted to the specialized nature of these classes. 

In this context, I'm wondering if it would be worth to make SetInputTexture abstract to force contributors to always use a more specialized subclasses, creating new ones if necessary.